### ISSUE TITLE
Update compatibility matrix for Treelite/XGBoost version support

### ISSUE THREAD

**Issue by wphicks** [triton-inference-server/fil_backend#412]
[This compatibility matrix](https://github.com/triton-inference-server/fil_backend/blob/main/docs/model_support.md#version-compatibility) should be updated to include UBJSON and reflect any other useful information since the release of 24.03

**Comment by whittenator** [comment]
Does the FIL backend support UBJSON format? If so, what parameters do you need to use in your config.pbtxt?

**Comment by hcho3** [comment]
Yes, starting from version 24.10, the FIL backend supports UBJSON format. As shown in [the doc](https://github.com/triton-inference-server/fil_backend/blob/main/docs/model_config.md), please set `model_type` to `xgboost_ubj`.
